src/Data/Pool/Internal.hs

```diff
@@ -171,17 +171,38 @@ destroyAllResources pool = forM_ (localPools pool) $ \lp -> do
 -- Helpers
 
 -- | Get a local pool.
-getLocalPool :: Maybe Int -> SmallArray (LocalPool a) -> IO (LocalPool a)
-getLocalPool mNumStripes pools = do
-  sid <- case mNumStripes of
-    -- If the number of stripes was set automatically to the number of
-    -- capabilities, we pick it based on the capability the thread currently
-    -- executes on. This ensures no contention on a specific stripe by
-    -- construction.
-    Nothing -> fmap fst . threadCapability =<< myThreadId
-    Just 1  -> pure 0
-    Just _  -> hash <$> myThreadId
-  pure $ pools `indexSmallArray` (sid `rem` sizeofSmallArray pools)
+getLocalPool :: SmallArray (LocalPool a) -> IO (LocalPool a)
+getLocalPool pools = do
+  let stripeCount = sizeofSmallArray pools
+  capabilities <- getNumCapabilities
```

Comment:
We get the number of capabilities directly. This is preferable to inspecting the configuration, since someone may have called `setNumCapabilities` after creating the pool.